core/trino-main/src/main/java/io/trino/sql/planner/NodePartitioningManager.java

```diff
@@ -83,6 +85,11 @@ public PartitionFunction getPartitionFunction(
                 .orElseThrow(() -> new IllegalArgumentException("Bucket to partition must be set before a partition function can be created"));
 
         PartitioningHandle partitioningHandle = partitioningScheme.getPartitioning().getHandle();
+
+        if (isScaledWriterPartitioningHandle(partitioningHandle)) {
```

Comment:
At this point it just seems that
```
boolean isScaled = partitioningHandle.getConnectorHandle() instanceOf ScaledWriterPartitioningHandle
```
but with an extra quirk that one now has to unwrap `getConnectorHandle()`, so that other instance checks work.

It makes me think that we should just add
```
boolean isScaled
```
to `PartitioningHandle` itself, because `ScaledWriterPartitioningHandle` just acts as a boolean.